src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -94,15 +95,36 @@ public final void doPost(HttpServletRequest request, HttpServletResponse respons
         response.sendError(403, ERROR_403);
         return;
       }
+      String userEmail = getUserEmail(request);
       Credential googleCredential = getGoogleCredential(request);
-      doPost(request, response, googleCredential);
+      doPost(request, response, googleCredential, userEmail);
     } catch (CredentialVerificationException e) {
       throw new ServletException(e.getMessage(), e);
     } catch (GeneralSecurityException e) {
       throw new ServletException(ERROR_500, e);
     }
   }
 
+  /**
+   * Handle GET request. Only override this method if the servlet needs access to the user's email
+   * Public for testing purposes
+   *
+   * @param request HTTP request from client
+   * @param response Http response to be sent to client
+   * @param googleCredential valid, verified google credential object
+   * @param userEmail the email address of the user
+   * @throws IOException if a read/write issue arises while processing the request
+   * @throws ServletException if the request cannot be handled due to unexpected errors
+   */
+  public void doGet(
+      HttpServletRequest request,
+      HttpServletResponse response,
+      Credential googleCredential,
+      String userEmail)
```

Comment:
Just curious - where do you use this parameter?

Reply:
It's not in this PR - it's a dependent PR (PR #139) as this is needed to check if the user was personally sent the email or if it was sent as a part of a mailing list